javascript/javascript1/week4/voiceAssistant.js

```diff
@@ -0,0 +1,84 @@
+let name = "";
+let toDoList = [];
+
+function getReply(command){
+    // lowercase command and split it into commandWords array
+   const commandWords = command.toLowerCase().split(" ");
+   let response = null;
+   let toDoText = "";
+
+   if (name !== "" && commandWords[0] == "hello"){
+    console.log("you've already introduced yourself") 
+    return 
+   }
+
+   switch(commandWords[0]) {
+      case "hello":
+        name = commandWords.pop();
+        response = "nice to meet you " + name; break;
+
+      case "what":
+        switch(true) {
+            case commandWords.includes("name"):{
+              if (name === ""){
+                response = "sorry! you have not told us your name yet!"; break;
+              }
+              response = `your name is ${name} `; break;}
+            case commandWords.includes("todo?"):
+              response = toDoList; break;
+            case commandWords.includes("today?"):
+              response = new Date().toDateString(); break;
+            case commandWords[3] == "+":
+              response = parseInt(commandWords[2]) + parseInt(commandWords[4]); break;
+            case commandWords[3] == "*":
+              response = commandWords[2] * commandWords[4]; break;
+            case commandWords[3] == "/":
+              response = commandWords[2] / commandWords[4]; break;
+            case commandWords[3] == "-":
+              response = commandWords[2] - commandWords[4]; break;
+        }
+        break;
+
+      case "add":
+          
+          for (let i = 1; i < commandWords.length - 3; i++){
+            toDoText += commandWords[i] + " ";
+          }
+          toDoList.push(toDoText);
+         response = toDoText + " added to your todo"; break;
+
+      case "remove":
+        
+        for (let i = 1; i < commandWords.length - 3; i++){
+          toDoText += commandWords[i] + " ";
+        }
+        toDoList.pop(toDoText);
+        response = "Removed " + toDoText + " from your todo"; break;
+
```

Comment:
`.pop` doesn't take a parameter like this.
If you want to find and remove an element you have to either use `.splice` or `.filter`.

Reply:
ok changed it.